### ISSUE TITLE
Corcom's dgun vs Armcom's

### ISSUE THREAD

**Issue by Jazcash** [beyond-all-reason/Beyond-All-Reason#2136]
cor's com dgun is very broken compared to arm's. It has a lot of problems when issuing dgun orders close to its max radius, which can cause it to either:

- get stuck and not fire at all
- move a bit and jostle around before firing
- fire ok, but then for some reason moves to stand on top of where dgun was clicked, with no visible move order

https://github.com/beyond-all-reason/Beyond-All-Reason/assets/1434248/bda952f0-9b30-4bdf-8fa7-e154fe1cea24

https://clips.twitch.tv/ExquisiteImportantKeyboardPhilosoraptor-N2qJpw5QS3xMynTJ

**Comment by WatchTheFort** [comment]
Appears to be a duplicate of https://github.com/beyond-all-reason/Beyond-All-Reason/issues/1900?

**Comment by Jazcash** [comment]
Imo this is the worst bug in BAR atm, very annoying

**Comment by StanczakDominik** [comment]
Doesn't make it any less of a duplicate ;) And it's blocked on the new engine, IIRC. That should be coming soon-ish!

**Comment by Avaristimo** [comment]
I was able to replicate both 1. and 3. for armcom.
While having experienced issue 2. ingame, I was not able to replicate it on either in tests.

I do note that the aiming is a tiny bit slower for corcom, upon checking, the corcom arm script moves at 300 instead of 390 for armcom.
Because the corcom arm is shorter, the dgun has 12 less range.

If that's OK with everyone, I can adjust both of these.

**Comment by RebelNode** [comment]
I did some testing for this. From my testing it looks like this happens when:

- Using cortex commander
- Aiming close to max range
- Aiming behind terrain (a bump, a cliff etc.)

Arm com is fine and isn't affected.

![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/1857760/6cb8479b-38ff-4c35-b492-665e50aa94ee)

**Comment by RebelNode** [comment]
> Because the corcom arm is shorter, the dgun has 12 less range.

If you can replicate the behaviour in my screenshot (it's from the middle island in Erebos Lakes map) then maybe try giving the cortex com a bit longer **arm** (heh) and see if that fixes the behaviour.

**Comment by Avaristimo** [comment]
[Screen Recording (09-01-2024 14-45-31).webm](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/73989722/46a0c754-6368-4bd6-a3a0-6a2c427be6b9)
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/73989722/47393873-884e-42cb-bdea-4552afc8bc9b)

Here's armada having the same issues.

**Comment by RebelNode** [comment]
I can only reproduce with corcom and only on uneven ground. It thinks it is in range of target so it doesn't walk forward, it just gets stuck trying to dgun a target that's out of range. If I use another unit to push it forward a bit then it will execute dgun and get unstuck.

**Comment by KyleAnthonyShepherd** [comment]
Deeply cursed engine problem, to my knowledge. Root problem is the arms animating before firing, potentially pulling the aimfrom and query points out of range. 

And the engine not correctly handling that.
